app/models/application_record.rb

```diff
@@ -34,22 +34,17 @@ def read_attribute_for_validation(attribute)
   def self.most_recently_changed(*record_classes)
     queries = record_classes.map do |clz|
       column_name = clz.send(:timestamp_attributes_for_update_in_model)&.first || "updated_at"
-      "(SELECT MAX(#{column_name}) AS max_updated_at FROM #{clz.table_name})"
+      table = clz.arel_table
+      table.project(table[column_name].maximum.as("max_updated_at")).to_sql
     end
       .join(" UNION ")
 
-    union_query = <<~SQL
-      SELECT MAX(union_query.max_updated_at)
+    union_query = <<~SQL.squish
+      SELECT MAX(max_updated_at)
       FROM (#{queries})
-      AS union_query
```

Comment:
I needed to readd this line locally to stop PG complaining about subqueries needing an alias.

Reply:
Good catch, I accidentally stayed on 16 since some tests not long ago, and exactly in version 16 requirement for naming subquery was removed, will switch back to 13